src/components/views/rooms/wysiwyg_composer/hooks/useEditing.ts

```diff
@@ -45,11 +45,12 @@ export function useEditing(
         [initialContent],
     );
 
-    const editMessageMemoized = useCallback(
-        () =>
-            !!mxClient && content !== undefined && editMessage(content, { roomContext, mxClient, editorStateTransfer }),
-        [content, roomContext, mxClient, editorStateTransfer],
-    );
+    const editMessageMemoized = useCallback(async () => {
+        if (mxClient === undefined || content === undefined) {
+            return;
+        }
+        return editMessage(content, { roomContext, mxClient, editorStateTransfer });
+    }, [content, roomContext, mxClient, editorStateTransfer]);
```

Comment:
This was previously returning `false | Promise<...>` which I don't think is what we were aiming for here.